src/BasicAccess.php

```diff
@@ -53,15 +53,31 @@ public function __construct(
     public function authenticate(ServerRequestInterface $request) : ?UserInterface
     {
         $authHeader = $request->getHeader('Authorization');
-        if (empty($authHeader)) {
+        if (! isset($authHeader[0])) {
             return null;
         }
 
-        if (! preg_match('/Basic (?P<credentials>[a-zA-Z0-9\+\/\=]+)/', $authHeader[0], $match)) {
+        if (! preg_match('/Basic (?P<credentials>.+)/', $authHeader[0], $match)) {
```

Comment:
I think its not good to maintain a list of valid base64 character in the regexp, because this list is also already maintained in PHP itself and checked with the strict argument in `base64_decode()`

Reply:
Don't assume PHP is checking the header values. The code may be run under different SAPIs or alternative process managers that may or may not do such checks (e.g. Swoole, ReactPHP).

Reply:
Can you elaborate what you mean exactly here? Are you implying, that the strict argument from `base64_decode()` may be a no-op in some environments?